### PR TITLE
Add breaking change "Passing generic type to RouteFactory"

### PR DESCRIPTION
**Pull request by AlexVincent525**
## Related pull request

https://github.com/flutter/flutter/pull/53390

### CODE REVIEW

File: src/docs/release/breaking-changes/passing-generic-type-to-route-factory.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+## Migration guide
+
+You must migrate your code as described in this section
+if you're subclassing or implementing `onGenerateRoute` / `onUnknownRoute`.
+If you do, the analyzer shows the following warnings when you
+upgrade to the Flutter version that includes this change:
+
+```none
+error • The argument type 'MaterialPageRoute<dynamic> Function(RouteSettings)' can't be assigned to the parameter type 'Route<T> Function<T>(RouteSettings)' • lib/main.dart:114:7 • argument_type_not_assignable
```
Comment (Hixie) [inline src/docs/release/breaking-changes/passing-generic-type-to-route-factory.md]: is there a cupertino version of this error too?

**Comment by sfshaza2** [comment]
The [PR](https://github.com//flutter/flutter/pull/53390) for this breaking change hasn't landed yet, so I will hold off on publishing.

File: src/docs/release/breaking-changes/passing-generic-type-to-route-factory.md (hunk 1 above)
Comment (AlexVincent525) [reply src/docs/release/breaking-changes/passing-generic-type-to-route-factory.md]: Yes it does, should I put it here too?

Comment (sfshaza2) [reply src/docs/release/breaking-changes/passing-generic-type-to-route-factory.md:26]: @AlexVincent525, not @Hixie, but if he wondered, others will, too.

**Comment by AlexVincent525** [comment]
As @dnfield suggest, this implementation will come up as a soft break, so I'm closing this since it's not needed for now.

**Pull request closed without merging**